zwave_js_server/client.py

```diff
@@ -199,181 +181,119 @@ async def async_send_json_message(self, message: Dict[str, Any]) -> None:
             )
-            self.tries = 0
 
             self.version = version = VersionInfo.from_message(
                 await client.receive_json()
             )
+        except (
+            client_exceptions.WSServerHandshakeError,
+            client_exceptions.ClientError,
+        ) as err:
+            raise CannotConnect(err) from err
+
+        # basic check for server version compatability
+        self._check_server_version(self.version.server_version)
+
+        self._logger.info(
+            "Connected to Home %s (Server %s, Driver %s)",
+            version.home_id,
+            version.server_version,
+            version.driver_version,
+        )
+        self.state = STATE_CONNECTED
 
-            # basic check for server version compatability
-            self._check_server_version(self.version.server_version)
-
-            self._logger.info(
-                "Connected to Home %s (Server %s, Driver %s)",
-                version.home_id,
-                version.server_version,
-                version.driver_version,
+        if self._on_connect:
+            asyncio.create_task(
+                gather_callbacks(self._logger, "on_connect", self._on_connect)
             )
-            self.state = STATE_CONNECTED
-
-            if self._on_connect:
-                asyncio.create_task(
-                    gather_callbacks(self._logger, "on_connect", self._on_connect)
-                )
-
-            loop = asyncio.get_running_loop()
-
-            while not client.closed:
-                msg = await client.receive()
 
-                if msg.type in (WSMsgType.CLOSED, WSMsgType.CLOSING):
-                    break
-
-                if msg.type == WSMsgType.ERROR:
-                    disconnect_warn = "Connection error"
-                    break
-
-                if msg.type != WSMsgType.TEXT:
-                    disconnect_warn = "Received non-Text message: {}".format(msg.type)
-                    break
-
-                try:
-                    if len(msg.data) > SIZE_PARSE_JSON_EXECUTOR:
-                        msg = await loop.run_in_executor(None, msg.json)
-                    else:
-                        msg = msg.json()
-                except ValueError:
-                    disconnect_warn = "Received invalid JSON."
-                    break
+    async def listen(self) -> None:
+        """Start listening to the websocket."""
+        assert self.client
 
-                if self._logger.isEnabledFor(logging.DEBUG):
-                    self._logger.debug("Received message:\n%s\n", pprint.pformat(msg))
+        loop = asyncio.get_running_loop()
 
-                msg_ = cast(dict, msg)
-                try:
-                    self.async_handle_message(msg_)
+        while not self.client.closed:
+            try:
+                msg = await self.client.receive()
+            except (
+                client_exceptions.WSServerHandshakeError,
+                client_exceptions.ClientError,
```

Comment:
What client error do you expect here?

Reply:
The docs doesn't mention any exceptions for the receive method. I think the message types ERROR will represent errors.

So I think we can remove the try... except.

Reply:
The only thing that can be raised here is CancelledError